src/NServiceBus.StructureMap/StructureMapDiagnostics.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace NServiceBus.Features
+{
+    /// <summary>
+    /// Adds Diagnostics information
+    /// </summary>
+    public class StructureMapDiagnostics : Feature
+    {
+        /// <summary>
+        /// Constructor for diagnostics feature
+        /// </summary>
+        public StructureMapDiagnostics()
+        {
+            EnableByDefault();
+        }
+
+        /// <summary>
+        /// Sets up diagnostics
+        /// </summary>
+        protected override void Setup(FeatureConfigurationContext context)
```

Comment:
is there no place this can be done that allows a new type to be avoided?

Reply:
Yes, I think this should be done here

https://github.com/Particular/NServiceBus.StructureMap/blob/develop/src/NServiceBus.StructureMap/StructureMapBuilder.cs#L25


Thoughts?

Reply:
@andreasohlund that would be where i would have assumed

Reply:
This would have the added benefit of only adding this if the container is a actually activated using `.UseContainer<T>()`

Reply:
I've moved it to the suggested location. We can always move it back to a new type if we need to add additional diagnostics at a later stage